analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/Ccsh.kt

```diff
@@ -75,15 +87,80 @@ class Ccsh : Callable<Void?> {
         fun executeCommandLine(args: Array<String>): Int {
             val commandLine = CommandLine(Ccsh())
             commandLine.executionStrategy = CommandLine.RunAll()
-            if (args.isEmpty() || isParserUnknown(args, commandLine)) {
-                return executeInteractiveParser(commandLine)
+            return if (args.isEmpty()) {
+                val configuredParsers = offerInteractiveParserSuggestions(commandLine)
+                if (configuredParsers.isEmpty()) {
+                    return 0
+                }
+
+                return executeConfiguredParsers(commandLine, configuredParsers)
+            } else if (isParserUnknown(args, commandLine) || args.contains("--interactive") || args.contains("-i")) {
+                executeInteractiveParser(commandLine)
             } else {
-                return commandLine.execute(*sanitizeArgs(args))
+                commandLine.execute(*sanitizeArgs(args))
             }
         }
 
+        fun offerInteractiveParserSuggestions(commandLine: CommandLine): Map<String, List<String>> {
+            val inputFile: String = KInquirer.promptInput(
+                    message = "What is the name of the input file/folder/url?",
```

Comment:
I would Maybe do it like this:
Which path should be scanned?
Hint: You can provide a directory path / file path / sonar url

Reply:
Sure that sounds better!